src/TwigExtension.php

```diff
@@ -487,32 +487,49 @@ public function searchApiQuery(string $index, string $term, array $fulltext, arr
         }
       }
       /* Basic is good enough for facets */
-      $query->setProcessingLevel(QueryInterface::PROCESSING_BASIC);
+      $query->setProcessingLevel(QueryInterface::PROCESSING_FULL);
       // see also \Drupal\search_api_autocomplete\Plugin\search_api_autocomplete\suggester\LiveResults::getAutocompleteSuggestions
       $results = $query->execute();
       $extradata = $results->getAllExtraData();
       // We return here
       $return = [];
-      foreach( $results->getResultItems() as $resultItem) {
-        $return['results'][$resultItem->getOriginalObject()->getValue()->id()]['entity'] = $resultItem->getOriginalObject()->getValue();
-       foreach ($resultItem->getFields() as $field) {
-         $return['results'][$resultItem->getOriginalObject()->getValue()->id()]['fields'][$field->getFieldIdentifier()] = $field->getValues();
-       }
+      foreach( $results->getResultItems() as $itemid => $resultItem) {
+        try {
+          $result_object = $resultItem->getOriginalObject(TRUE);
+          if ($result_object instanceof \Drupal\Core\Entity\Plugin\DataType\EntityAdapter) {
+            $return['results'][$itemid]['entity']
+              = $result_object->getEntity();
+          }
+          else {
+            $return['results'][$itemid]['entity'] = NULL;
+          }
+        }
+        catch (SearchApiException $e) {
+          $return['results'][$itemid]['entity'] = NULL;
+        }
+
+        if ($result_object) {
+          foreach ($resultItem->getFields() as $field) {
+            $return['results'][$itemid]['fields'][$field->getFieldIdentifier()]
```

Comment:
Yes, because we can't load the Entity to get the real ID. We could split it upfront to get the entity but that said having a way to letting the user know (via the ID) what entity was fetched and the language might be good (means entity:node/15:en seems a way of doing that)